src/zx/FunctionalityConstruction.cpp

```diff
@@ -223,6 +226,65 @@ FunctionalityConstruction::parseOp(ZXDiagram& diag, op_it it, op_it end,
       addZSpider(diag, target2, qubits, PiExpression(), EdgeType::Hadamard);
       break;
     }
+    case qc::OpType::RZZ: {
+      const auto ctrl = target;
+      const auto target2 = static_cast<zx::Qubit>(p.at(op->getTargets()[1]));
+      const auto param = parseParam(op.get(), 0);
+
+      addZSpider(diag, ctrl, qubits);
+      addZSpider(diag, target2, qubits);
+
+      auto mid_x = diag.addVertex(-1, -1, PiExpression(PiRational(0, 1)),
```

Comment:
I know I am probably just picky again but it seems kind of strange to me that you "have" to add an artificial PiExpression that is just 0 here.
Wouldn't it be way nicer interface-wise, that I could just say `diag.addXVertex()` or something similar like that?

Reply:
I see all your points but I only provided this PR so that Aaron could compare against ZX for certain Ising models. I don't want this to turn into a clean-up of the ZX library.

Reply:
Agreed. Could we still note that down somewhere so it is not lost in the void?